libs/docs/src/docs/2.2.0/innerloop-vs-outerloop.md

```diff
@@ -0,0 +1,420 @@
+
+    ```yaml
+    commands:
+      - id: build-image
+        apply:
+          component: outerloop-build
+      - id: deploy-openshift
+        apply:
+          component: outerloop-deploy
+      - id: deploy
+        composite:
+          commands:
+            - build-image
+            - deploy-openshift
+          group:
+            kind: deploy
+            isDefault: true
+    ```
+
+```yaml {% title="Final outerloop devfile" filename="devfile.yaml" %}
+schemaVersion: 2.2.0
+metadata:
+  name: nodejs
+components:
+  - name: outerloop-build
+    image:
+      imageName: landingpage-image:latest
+      dockerfile:
+        uri: docker/Dockerfile
+        buildContext: .
+        rootRequired: false
+  - name: outerloop-deploy
+    openshift:
+      uri: landingpage-template.yaml
+commands:
+  - id: build-image
+    apply:
+      component: outerloop-build
+  - id: deploy-openshift
+    apply:
+      component: outerloop-deploy
+  - id: deploy
+    composite:
+      commands:
+        - build-image
+        - deploy-openshift
+      group:
+        kind: deploy
+        isDefault: true
+```
+
+With these components and commands, the developer can produce a full build and deployment. In this case, the project has an OpenShift deployment template `landingpage-template.yaml` for a full deploy and docker image build file `docker/Dockerfile` for a full build. These actions can also be used for the requirements of performing integration testing.
+
+## Support list of developer tools
+
+Though the devfile spec does support both [innerloop](#what-is-innerloop) and [outerloop](#what-is-outerloop) deployments, not all developer tools will support these deployments in the same way. This section will list the devfile supported development tools and what deployment scopes they support.
+
+### Developer Tools
+
+| Tool | Innerloop Support | Outerloop Support |
+|------|-------------------|-------------------|
+| Odo v2 | X | X |
+| Odo v3 | X | X |
+| Eclipse Che | X | |
+| Amazon CodeCatalyst | X | |
+| JetBrains Space Cloud Dev | X | X (*image build only*) |
+| Red Hat OpenShift Dev Spaces | X | |
+| OpenShift Dev Console | | X |
+| VSCode OpenShift Toolkit | X | X |
+| IntelliJ OpenShift Toolkit | X | X |
+
+## Odo Spec Support
+
+Odo covers the features supported by the devfile 2.2.0 spec with minor requirements for version 3: 
+1. In order to debug your project it is required to define a debug port: [https://odo.dev/docs/user-guides/v3-migration-guide#changes-to-the-way-component-debugging-works](https://odo.dev/docs/user-guides/v3-migration-guide#changes-to-the-way-component-debugging-works)
+2. By default, odov3 sets up an underlying persistent volumes rather than setting up ephemeral volumes, this can toggled back: [https://odo.dev/docs/user-guides/v3-migration-guide#ephemeral-storage](https://odo.dev/docs/user-guides/v3-migration-guide#ephemeral-storage)
```

Comment:
```suggestion
2. By default, odo v3 sets up underlying persistent volumes rather than setting up ephemeral volumes, this can be toggled back: [https://odo.dev/docs/user-guides/v3-migration-guide#ephemeral-storage](https://odo.dev/docs/user-guides/v3-migration-guide#ephemeral-storage)
```